light-node/src/commands/start.rs

```diff
@@ -0,0 +1,159 @@
+//! `start` subcommand - start the light node.
+
+/// App-local prelude includes `app_reader()`/`app_writer()`/`app_config()`
+/// accessors along with logging macros. Customize as you see fit.
+use crate::prelude::*;
+
+use tendermint::hash;
+use tendermint::lite;
+use tendermint::lite::ValidatorSet as _;
+use tendermint::lite::{Header, Height, Requester, TrustThresholdFraction};
+use tendermint::rpc;
+use tendermint::Hash;
+
+use crate::application::APPLICATION;
+use crate::config::LightNodeConfig;
+use crate::requester::RPCRequester;
+use crate::store::{MemStore, State};
+use abscissa_core::{config, Command, FrameworkError, Options, Runnable};
+use std::process;
+use std::time::{Duration, SystemTime};
+use tendermint::lite::error::Error;
+
+/// `start` subcommand
+///
+/// The `Options` proc macro generates an option parser based on the struct
+/// definition, and is defined in the `gumdrop` crate. See their documentation
+/// for a more comprehensive example:
+///
+/// <https://docs.rs/gumdrop/>
+#[derive(Command, Debug, Options)]
+pub struct StartCmd {
+    /// RPC address to request headers and validators from.
+    #[options(free)]
+    rpc_addr: String,
+}
+
+impl Runnable for StartCmd {
+    /// Start the application.
+    fn run(&self) {
+        if let Err(err) = abscissa_tokio::run(&APPLICATION, async {
```

Comment:
Perhaps add a comment here about what this does? I assume spawning a task to run the async block? What kind of runtime does it start, threaded or current thread? If threaded, how many? I'd say we just need a basic scheduler for now as in https://docs.rs/tokio/0.1.20/tokio/runtime/current_thread/index.html

Reply:
The runtime is started as an Abscissa component. You can use whatever runtime you'd like... see the `From<Runtime>` impl on `TokioComponent`:

https://docs.rs/abscissa_tokio/0.5.1/abscissa_tokio/struct.TokioComponent.html

Right now it's using the default, which is equivalent to `tokio::runtime::Runtime::new()`:

https://github.com/interchainio/tendermint-rs/pull/125/files#diff-e4205fba6a4d881725a2a98ea6858b7aR87

By using an Abscissa component to manage the runtime, other components can spawn additional tasks on the reactor or leverage other integrations (e.g. [tracing](https://github.com/tokio-rs/tracing), which Zebra is using extensively)